src/main/java/org/jabref/gui/entryeditor/citationrelationtab/RelatedEntriesComponent.java

```diff
@@ -0,0 +1,237 @@
+package org.jabref.gui.entryeditor.citationrelationtab;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.swing.undo.UndoManager;
+
+import javafx.collections.FXCollections;
+import javafx.css.PseudoClass;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.control.Tooltip;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.LibraryTab;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.icon.IconTheme;
+import org.jabref.gui.util.NoSelectionModel;
+import org.jabref.gui.util.ViewModelListCellFactory;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.util.FileUpdateMonitor;
+import org.jabref.preferences.PreferencesService;
+
+import com.tobiasdiez.easybind.EasyBind;
+import org.controlsfx.control.CheckListView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RelatedEntriesComponent extends VBox {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RelatedEntriesComponent.class);
+    private final LibraryTab libraryTab;
+    private final BibEntry pivotEntry;
+    private final RelatedEntriesComponentConfig config;
+    private final RelatedEntriesComponentViewModel viewModel;
+    private Button refreshButton;
+    private Button cancelButton;
+    private ProgressIndicator progressIndicator;
+    private Button importEntriesButton;
+    private CheckListView<CitationRelationItem> relatedEntriesListView;
+
+    public RelatedEntriesComponent(BibEntry pivotEntry, RelatedEntriesRepository repository, LibraryTab libraryTab,
+                                   RelatedEntriesComponentConfig config, DialogService dialogService,
+                                   BibDatabaseContext databaseContext, UndoManager undoManager,
+                                   StateManager stateManager, FileUpdateMonitor fileUpdateMonitor,
+                                   PreferencesService prefs) {
+        this.pivotEntry = pivotEntry;
+        this.config = config;
+        this.viewModel = new RelatedEntriesComponentViewModel(pivotEntry, repository, dialogService, databaseContext,
+                undoManager, stateManager, fileUpdateMonitor, prefs);
+        this.libraryTab = libraryTab;
+
+        initialize();
+        if (canLoadEntries()) {
+            viewModel.loadEntries();
+        } else {
+            relatedEntriesListView.setPlaceholder(buildLabel(Localization.lang("The selected entry doesn't have a DOI linked to it. Lookup a DOI and try again.")));
```

Comment:
Please use `does not` instead of `doesn't`

Reply:
Done.